ql/src/java/org/apache/hadoop/hive/ql/exec/repl/util/ReplUtils.java

```diff
@@ -474,4 +475,17 @@ public static void addLoggerTask(ReplLogger replLogger, List<Task<?>> tasks, Hiv
       DAGTraversal.traverse(tasks, new AddDependencyToLeaves(Collections.singletonList(task)));
     }
   }
+
+  public static void reportStatusInReplicationMetrics(String stageName, Status status, String errorLogPath,
+                                                      HiveConf conf)
+          throws SemanticException {
+    GenericMetricCollector metricCollector = new GenericMetricCollector(0, conf);
```

Comment:
What advantage is ``GenericMetricCollector`` class giving you?
It is just calling the super constructor with pre-set value, that you could do it here itself. Or by having a Constructor in the parent class which takes only two params and set other values as ``null``

Reply:
Done